src/components/AppBrowser.js

```diff
@@ -1,42 +1,55 @@
 import QuickId from '@/components/QuickId.vue'
 import { AeAppIcon } from '@aeternity/aepp-components'
-
+import { PATHS } from '../router'
 export default {
   name: 'app-browser',
   data () {
     return {
       iframe: '',
-      showIframe : false,
-      iframeLoading : true,
+      showIframe: false,
+      iframeLoading: true
     }
   },
-  computed : {
-    apps() {
+  watch: {
+    url (newUrl, oldUrl) {
+      this.resolveUrl(newUrl)
+    },
+    iframe (newIframe, oldIframe) {
+      // replace the location so it will not be added to the histoty stack of the iframe
+      this.$refs.appframe.contentWindow.location.replace(this.iframe)
+    }
+  },
+  computed: {
+    url () {
+      return this.$route.query && this.$route.query.aepp
+    },
+    apps () {
       return this.$store.state.apps
     },
-    iframeStyle() {
+    iframeStyle () {
```

Comment:
Styling should be part of the Js Code. We should set a CSS (modifier) class depending on the existence of the iframe and do the actual styling withing the CSS code.